nomos-services/mempool/src/processor/tx.rs

```diff
@@ -0,0 +1,83 @@
+use std::error::Error;
+
+use futures::{stream::FuturesUnordered, StreamExt};
+use nomos_core::{
+    da::BlobId,
+    mantle::{ops::Op, tx::SignedMantleTx},
+};
+use nomos_da_sampling::DaSamplingServiceMsg;
+use overwatch::services::{relay::OutboundRelay, ServiceData};
+
+use super::PayloadProcessor;
+
+#[derive(thiserror::Error, Debug)]
+pub enum SignedTxProcessorError {
+    #[error("Error from sampling relay {0}")]
+    Sampling(Box<dyn Error + Send>),
+}
+
+impl SignedTxProcessorError {
+    fn sampling_error(err: impl Error + Send + 'static) -> Self {
+        Self::Sampling(Box::new(err))
+    }
+}
+
+pub struct SignedTxProcessor<SamplingService>
+where
+    SamplingService: ServiceData,
+{
+    sampling_relay: OutboundRelay<<SamplingService as ServiceData>::Message>,
+}
+
+#[async_trait::async_trait]
+impl<SamplingService> PayloadProcessor for SignedTxProcessor<SamplingService>
+where
+    SamplingService: ServiceData<Message = DaSamplingServiceMsg<BlobId>> + Send + Sync,
+{
+    type Payload = SignedMantleTx;
+    type Settings = ();
+    type Error = SignedTxProcessorError;
+
+    type DaSamplingService = SamplingService;
+
+    fn new(
+        (): Self::Settings,
+        sampling_relay: OutboundRelay<<Self::DaSamplingService as ServiceData>::Message>,
+    ) -> Self {
+        Self { sampling_relay }
+    }
+
+    async fn process(&self, payload: &Self::Payload) -> Result<(), Vec<Self::Error>> {
+        let all_futures: FuturesUnordered<_> = payload
+            .mantle_tx
+            .ops
+            .iter()
+            .filter_map(|op| {
+                if let Op::Blob(blob_op) = op {
+                    Some(async {
+                        self.sampling_relay
+                            .send(DaSamplingServiceMsg::TriggerSampling {
+                                blob_id: blob_op.blob,
+                            })
+                            .await
+                            .map_err(|(e, _)| SignedTxProcessorError::sampling_error(e))
+                    })
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        let errors: Vec<SignedTxProcessorError> = StreamExt::collect::<Vec<_>>(all_futures)
```

Comment:
does this wait for the sampling result? I think a comment indicating whether it does or not would be helpful

Reply:
Added a comment https://github.com/logos-co/nomos/pull/1367/commits/8737708b9508ce1c4b4931c1c2dd7f8dc8068b04
```
        // In this case errors are about sending the `TriggerSampling` message to the DA
        // Sampling Service. These errors do not represent the result of
        // sampling itself, but might indicate that sending the message to
        // sampling service failed.
```